src/components/Header/AddEditDialog/AddEditDialog.tsx

```diff
@@ -48,83 +47,89 @@ export const AddEditDialog = (props: AddEditDialogProps): JSX.Element => {
-              <MwInput
-                id="title"
-                className="mwMiddleInput"
-                name="title"
-                onChange={handleInputChange}
-                type="text"
-                value={movieItem?.title}
-              />
-              <MwInput
-                id="releaseDate"
-                className="mwShortInput"
-                name="Release date"
-                onChange={handleInputChange}
-                type="date"
-                value={movieItem?.releaseDate}
-              />
-            </div>
-            <div className="mwFormRow">
-              <MwInput
-                id="posterPath"
-                className="mwMiddleInput"
-                name="Movie URL"
-                onChange={handleInputChange}
-                type="text"
-                value={movieItem?.posterPath}
-                placeholder="https://"
-              />
-              <MwInput
-                id="voteAverage"
-                className="mwShortInput"
-                name="Rating"
-                onChange={handleInputChange}
-                type="string"
-                value={movieItem?.voteAverage.toString()}
-                placeholder="7.8"
-              />
-            </div>
-            <div className="mwFormRow">
-              <MwInput
-                id="genres"
-                className="mwMiddleInput"
-                name="Genre"
-                onChange={handleInputChange}
-                type="text"
-                value={movieItem?.genres.toString()}
-              />
-              <MwInput
-                id="runtime"
-                className="mwShortInput"
-                name="Runtime"
-                onChange={handleInputChange}
-                type="text"
-                value={movieItem?.runtime.toString()}
-                placeholder="minutes"
-              />
-            </div>
-            <div className="mwFormRow">
-              <MwInput
-                id="overview"
-                className="mwLongInput"
-                name="Overview"
-                onChange={handleInputChange}
-                type="textarea"
-                value={movieItem?.overview}
-                placeholder="movie description"
-              />
-            </div>
-          </form>
-        </div>
-        <div className="mwFormFooter">
-          <MwButton onClick={props.handleCancelClick} className="mwCancelButton" buttonName="Reset" />
-          <MwButton onClick={onSubmitClick} className="mwSubmitButton" buttonName="Submit" />
-        </div>
+        <Formik
+          initialValues={{
+            title: movieItem?.title,
+            posterPath: movieItem?.posterPath,
+            overview: movieItem?.overview,
+            genres: movieItem?.genres,
+            releaseDate: movieItem?.releaseDate,
+            voteAverage: movieItem?.voteAverage,
+            runtime: movieItem?.runtime,
+          }}
+          validationSchema={Yup.object({
```

Comment:
Could you please move validation object outside the component so it isn't recreated on each render?

Reply:
sure, thanks